### PR TITLE
Bug 1883847: Fix subscription name issue for OCS

### PR DESCRIPTION
**Pull request by a2batic**
Removes subscription name dependencies for feature detection in OCS

### CODE REVIEW

File: frontend/packages/ceph-storage-plugin/src/features.ts [hunk 1]
```diff
@@ -99,18 +103,12 @@ export const detectOCS: FeatureDetector = async (dispatch) => {
...
-      CEPH_STORAGE_NAMESPACE,
-    );
-    const ocsCSV = await fetchK8s(
-      ClusterServiceVersionModel,
-      subscription?.status?.currentCSV,
-      CEPH_STORAGE_NAMESPACE,
-    );
+    const ocsCSV = await fetchK8s(ClusterServiceVersionModel, '', CEPH_STORAGE_NAMESPACE);
+    // eslint-disable-next-line dot-notation
+    const ocsItems = ocsCSV['items'] as ClusterServiceVersionKind[];
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/src/features.ts:108]:
```suggestion
    const csvItems = ocsList.items as ClusterServiceVersionKind[];
```

File: frontend/packages/ceph-storage-plugin/src/features.ts [hunk 2]
```diff
@@ -99,18 +103,12 @@ export const detectOCS: FeatureDetector = async (dispatch) => {
...
+    const ocsObj = ocsItems.find((obj) => getName(obj).includes(OCS_OPERATOR));
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/src/features.ts:109]:
```suggestion
    const ocsCSV = csvItems.find((obj) => getName(obj).includes(OCS_OPERATOR));
```

File: frontend/packages/ceph-storage-plugin/src/features.ts [hunk 3]
```diff
@@ -99,18 +103,12 @@ export const detectOCS: FeatureDetector = async (dispatch) => {
...
 
-    const support = getAnnotations(ocsCSV)[OCS_SUPPORT_ANNOTATION];
+    const support = getAnnotations(ocsObj)[OCS_SUPPORT_ANNOTATION];
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/src/features.ts:111]:
```suggestion
    const support = getAnnotations(ocsCSV)[OCS_SUPPORT_ANNOTATION];
```

File: frontend/packages/ceph-storage-plugin/src/features.ts [hunk 4]
```diff
@@ -99,18 +103,12 @@ export const detectOCS: FeatureDetector = async (dispatch) => {
...
-      SubscriptionModel,
-      'ocs-subscription',
-      CEPH_STORAGE_NAMESPACE,
-    );
-    const ocsCSV = await fetchK8s(
-      ClusterServiceVersionModel,
-      subscription?.status?.currentCSV,
-      CEPH_STORAGE_NAMESPACE,
-    );
+    const ocsCSV = await fetchK8s(ClusterServiceVersionModel, '', CEPH_STORAGE_NAMESPACE);
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/src/features.ts:106]:
```suggestion
    const csvList = await fetchK8s(ClusterServiceVersionModel, '', CEPH_STORAGE_NAMESPACE);
```

**Review by bipuladh — CHANGES_REQUESTED** [review]

File: frontend/packages/ceph-storage-plugin/src/features.ts (hunk 1 above)
Comment (a2batic) [reply frontend/packages/ceph-storage-plugin/src/features.ts]: `csvItems` will not be an array by type, so using dot-notation will throw compile-time error.

File: frontend/packages/ceph-storage-plugin/src/features.ts [hunk 5]
```diff
@@ -99,16 +103,10 @@ export const detectOCS: FeatureDetector = async (dispatch) => {
...
-      CEPH_STORAGE_NAMESPACE,
-    );
-    const ocsCSV = await fetchK8s(
-      ClusterServiceVersionModel,
-      subscription?.status?.currentCSV,
-      CEPH_STORAGE_NAMESPACE,
-    );
+    const csvList = await fetchK8s(ClusterServiceVersionModel, '', CEPH_STORAGE_NAMESPACE);
+    // eslint-disable-next-line dot-notation
+    const csvItems = csvList['items'] as ClusterServiceVersionKind[];
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/src/features.ts:108]:
```suggestion
    const csvItems = csvList?.items as ClusterServiceVersionKind[];
```

**Review by bipuladh — COMMENTED** [review]
Why not do this?

File: frontend/packages/ceph-storage-plugin/src/features.ts (hunk 5 above)
Comment (a2batic) [reply frontend/packages/ceph-storage-plugin/src/features.ts:108]: `csvItems` will not be an array by type, so using dot-notation will throw compile-time error.

Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/src/features.ts:108]:
```suggestion
    const csvList: any = await fetchK8s(ClusterServiceVersionModel, '', CEPH_STORAGE_NAMESPACE);
    const csvItems = csvList.items as ClusterServiceVersionKind[];
```

**Review by bipuladh — CHANGES_REQUESTED** [review]

**Comment by bipuladh** [comment]
> @bipuladh: once the present PR merges, I will cherry-pick it on top of release-4.6 in a new PR and assign it to you.

Thank you

**Pull request merged**